kartothek/io_components/metapartition.py

```diff
@@ -1365,6 +1366,15 @@ def _partition_data(self, partition_on):
         ]
         dct = dict()
         empty_tables = []
+
+        performing_safety_check = False
+        if self.data:  # Safety check: did dataframe size change?
+            performing_safety_check = True
+            safety_check_table_name = list(self.data.keys())[0]
```

Comment:
Why only check for a single table? As long as we still support multi tables we should verify this for all.

Also, this check might not be deterministic since the order of the keys might change from partition to partition

Reply:
> we still support multi tables

😞 Can't wait until we drop support